graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/processor/EntityModel.java

```diff
@@ -149,6 +153,76 @@ public boolean isFederated() {
     return inputTypeName;
   }
 
+  /**
+   * Validates a set of conditions to ensure that they will produce a valid CQL query for this
+   * entity.
+   *
+   * @return an optional error message if the validation failed, empty otherwise.
+   */
+  public Optional<String> validateConditions(Collection<WhereConditionModel> conditions) {
+
+    // TODO revisit these rules when we allow regular columns
+
+    // All partition key fields must be restricted by equality relations
+    for (FieldModel field : this.getPartitionKey()) {
+      Optional<WhereConditionModel> maybeCondition =
+          conditions.stream().filter(c -> c.getField().equals(field)).findFirst();
+      if (!maybeCondition.isPresent()) {
+        return Optional.of(
+            String.format(
+                "every partition key field of type %s must be present (expected: %s).",
+                this.getGraphqlName(),
+                this.getPartitionKey().stream()
+                    .map(FieldModel::getGraphqlName)
+                    .collect(Collectors.joining(", "))));
+      }
+      WhereConditionModel condition = maybeCondition.get();
+      if (condition.getPredicate() != Predicate.EQ && condition.getPredicate() != Predicate.IN) {
+        return Optional.of(
+            String.format(
+                "the only predicates allowed for partition key fields are EQ and IN (got %s for %s).",
+                condition.getPredicate(), field.getGraphqlName()));
+      }
+    }
+
+    // When a clustering field is NOT restricted by an equality relation, then no other clustering
+    // field after it can be restricted by any condition.
+    String firstNonEqField = null;
+    for (FieldModel field : this.getClusteringColumns()) {
+      Optional<WhereConditionModel> maybeCondition =
+          conditions.stream().filter(c -> c.getField().equals(field)).findFirst();
+      if (!maybeCondition.isPresent()) {
+        if (firstNonEqField == null) {
```

Comment:
how do you know this is non `EQ` field, you are not checking the `Predicate` in the WhereConditionModel? Actually I am confused, you basically set firstNonEqField in case of a missing clustering column in the where, can you elaborate on this?

What if you do find a field, that has the `EQ` predicate?

Reply:
If we enter this branch, there is no condition at all. So it passes the test "is not restricted by EQ or IN".

In other words, given:
```
type Foo @cql_input {
  pk: Int! @cql_column(partitionKey: true
  cc1: Int! @cql_column(clusteringOrder: ASC)
  cc2: Int! @cql_column(clusteringOrder: ASC)
}
type Query {
  query1(
    pk: Int
    cc1: Int @cql_where(predicate: LT)
    cc2: Int
  ): [Foo]

  query2(
    pk: Int
    cc2: Int
  ): [Foo]
}
```
Both queries are invalid because they have a condition on `cc2` without a sufficient restriction of `cc1`.

Reply:
Understood. Will resolve this.